pkg/ingestor/parser/cdx_vex/parser_cdx_vex.go

```diff
@@ -0,0 +1,217 @@
+	"github.com/guacsec/guac/pkg/assembler/helpers"
+	"github.com/guacsec/guac/pkg/handler/processor"
+	"github.com/guacsec/guac/pkg/ingestor/parser/common"
+	"github.com/guacsec/guac/pkg/logging"
+)
+
+var vexStatusMap = map[cdx.ImpactAnalysisState]generated.VexStatus{
+	"resolved":     generated.VexStatusFixed,
+	"exploitable":  generated.VexStatusAffected,
+	"in_triage":    generated.VexStatusUnderInvestigation,
+	"not_affected": generated.VexStatusNotAffected,
+}
+
+var justificationsMap = map[cdx.ImpactAnalysisJustification]generated.VexJustification{
+	"code_not_present":   generated.VexJustificationVulnerableCodeNotPresent,
+	"code_not_reachable": generated.VexJustificationVulnerableCodeNotInExecutePath,
+}
+
+type cdxVexParser struct {
+	doc               *processor.Document
+	identifierStrings *common.IdentifierStrings
+	cdxBom            *cdx.BOM
+}
+
+func NewCdxVexParser() common.DocumentParser {
+	return &cdxVexParser{
+		identifierStrings: &common.IdentifierStrings{},
+	}
+}
+
+// Parse breaks out the document into the graph components
+func (c *cdxVexParser) Parse(ctx context.Context, doc *processor.Document) error {
+	c.doc = doc
+	err := json.Unmarshal(doc.Blob, &c.cdxBom)
+	if err != nil {
+		return fmt.Errorf("unable to unmarshal CDX VEX document: %w", err)
+	}
+
+	return nil
+}
+
+// GetIdentities gets the identity node from the document if they exist
+func (c *cdxVexParser) GetIdentities(ctx context.Context) []common.TrustInformation {
+	return nil
+}
+
+func (c *cdxVexParser) GetIdentifiers(ctx context.Context) (*common.IdentifierStrings, error) {
+	return nil, fmt.Errorf("not yet implemented")
+}
+
+// Get package name and range versions to create package input spec for the affected packages.
+func getAffectedPackages(ctx context.Context, vulnInput *generated.VulnerabilityInputSpec, vexData generated.VexStatementInputSpec, affectsObj cdx.Affects) *[]assembler.VexIngest {
+	logger := logging.FromContext(ctx)
+	var pkgRef string
+	// TODO: retrieve purl from metadata if present - https://github.com/guacsec/guac/blob/main/pkg/ingestor/parser/cyclonedx/parser_cyclonedx.go#L76
+	if affectsObj.Ref != "" {
+		pkgRef = affectsObj.Ref
+	} else {
+		logger.Warnf("[cdx vex] package reference not found")
+		return nil
+	}
+
+	// split ref using # as delimiter.
+	pkgRefInfo := strings.Split(pkgRef, "#")
+	if len(pkgRefInfo) != 2 {
+		logger.Warnf("[cdx vex] malformed package reference: %q", affectsObj.Ref)
+		return nil
+	}
+	pkgURL := pkgRefInfo[1]
+
+	// multiple package versions do not exist, resolve to using ref directly.
+	if affectsObj.Range == nil {
+		pkg, err := helpers.PurlToPkg(pkgURL)
+		if err != nil {
+			logger.Warnf("[cdx vex] unable to create package input spec: %v", err)
+			return nil
+		}
+
+		return &[]assembler.VexIngest{{VexData: &vexData, Vulnerability: vulnInput, Pkg: pkg}}
+	}
+
+	// split pkgURL using @ as delimiter.
+	pkgURLInfo := strings.Split(pkgURL, "@")
```

Comment:
would pkgUrlInfo[1] be also used as part of the lists iterated below or is it guaranteed to be part of affectsObj.Range

Reply:
A little unsure about this. Also the examples [here](https://github.com/CycloneDX/bom-examples/blob/master/VEX/CISA-Use-Cases/Case-8/vex.json) don't seem to have pkg url listed in the ref. Used [this](https://raw.githubusercontent.com/CycloneDX/bom-examples/master/VEX/vex.json) cdx as a reference for this PR.

Reply:
Will try digging into it and post the necessary changes to a follow-up PR if required.